src/main/resources/org/jenkinsci/plugins/workflow/job/console/NewNodeConsoleNote/script.js

```diff
@@ -1,41 +1,62 @@
-Behaviour.specify("span.pipeline-new-node", 'NewNodeConsoleNote', 0, function(e) {
-    if (e.processedNewNodeConsoleNote) {
-        return
-    }
-    e.processedNewNodeConsoleNote = true
-    var label = e.getAttribute('label')
+var enclosings = new Map() // id → enclosingId
+var labels = new Map() // id → label
+
+/**
+ * Fix label attributes
+ */
+function fixLabels(pipelineElement) {
+    var label = pipelineElement.getAttribute('label')
     if (label != null) {
-        var html = e.innerHTML
+        var html = pipelineElement.innerHTML
         var suffix = ' (' + label.escapeHTML() + ')';
         if (html.indexOf(suffix)<0) {
-            e.innerHTML = e.innerHTML.replace(/.+/, '$&' + suffix) // insert before EOL
+            pipelineElement.innerHTML = pipelineElement.innerHTML.replace(/.+/, '$&' + suffix) // insert before EOL
         }
-    }
-    var nodeId = e.getAttribute('nodeId')
-    var startId = e.getAttribute('startId')
-    if (startId == null || startId == nodeId) {
-        e.innerHTML = e.innerHTML.replace(/.+/, '$&<span class="pipeline-show-hide"> (<a href="#" onclick="showHidePipelineSection(this); return false">hide</a>)</span>')
-        // TODO automatically hide second and subsequent branches: namely, in case a node has the same parent as an earlier one
-    }
-    // The CSS rule for branch names only needs to be added once per node, so we
-    // check in case we are viewing the truncated log and have already processed
-    // a duplicate synthetic span element for this node.
-    var maybeDupeNodes = $$('[nodeid=\"'+nodeId+'\"].pipeline-new-node');
-    for (var i = 0; i < maybeDupeNodes.length; i++) {
-        var node = maybeDupeNodes[i];
-        if (node !== e && node.processedNewNodeConsoleNote) {
-            return;
-        }
-    }
-    var nodes = $$('.pipeline-new-node')
+     }
+}
+
+/**
+ * Generate hide/show links for all of the pipeline elements
+ */
+function addLinks (pipelineElement) {
+       var nodeId = pipelineElement.getAttribute('nodeId')
+       var startId = pipelineElement.getAttribute('startId')
+       if (startId == null || startId == nodeId) {
+            pipelineElement.innerHTML = pipelineElement.innerHTML.replace(/.+/, '$&<span class="pipeline-show-hide"> (<a href="#" onclick="showHidePipelineSection(this); return false">hide</a>)</span>')
+       }
+}
+
+/**
+ * Generate enclosings for a element
+ */
+function generateEnclosing (node) {
+    var oid = node.getAttribute('nodeId')
+    enclosings.set(oid, node.getAttribute('enclosingId'))
+    labels.set(oid, node.getAttribute('label'))
+}
+
+/**
+ * Generate enclosings and labels for all of the pipeline elements
+ */
+function generateEnclosings () {
     var enclosings = new Map() // id → enclosingId
     var labels = new Map() // id → label
```

Comment:
Did you mean to create a local version of `enclosings` and `labels` here, or were you wanting to reuse and modify the global variables and use them as a kind of cache or something?

The way things are now, it doesn't look like `generateEnclosing` actually does anything because the global variables aren't used.